src/components/chart/range-annotations-options.js

```diff
@@ -0,0 +1,223 @@
+            }
+        });
+        return {
+            xAxis: xAxisPlotBands.length > 0 ? { plotBands: xAxisPlotBands } : undefined,
+            yAxis: yAxisPlotBands.length > 0 ? { plotBands: yAxisPlotBands } : undefined,
+        };
+    };
+
+    // Returns the position of the label relative to the plotBand
+    getLabelPosition = (chartType, labelRect, plotBandRect, isXaxis) => {
+        let isToLeft = false;
+        let isToRight = false;
+        let isToTop = false;
+        let isToBottom = false;
+        let isVertical = chartType === 'bar' ? !isXaxis : isXaxis;
+        if (isVertical) {
+            // vertical plotBand
+            if (labelRect.right < plotBandRect.left) {
+                isToLeft = true;
+            }
+            if (labelRect.left > plotBandRect.right) {
+                isToRight = true;
+            }
+        } else {
+            // horizontal plotBand
+            if (labelRect.bottom < plotBandRect.top) {
+                isToTop = true;
+            }
+            if (labelRect.top > plotBandRect.bottom) {
+                isToBottom = true;
+            }
+        }
+        return { isToLeft, isToRight, isToTop, isToBottom };
+    };
+
+    // Draws the line connecting the label to the plotBand
+    // Also adds a custom class to the label based on its position
+    addLine = (currentChart) => {
+        // If there is already a line in the chart, remove it before redrawing it
+        const lines = currentChart.container.querySelectorAll('[data-range-annotation-line]');
+        if (lines.length > 0) {
+            lines.forEach((line) => {
+                line.remove();
+            });
+        }
+
+        // Get all plotBands from both axes
+        const xAxisPlotBands = currentChart.xAxis[0].plotLinesAndBands;
+        const yAxisPlotBands = currentChart.yAxis[0].plotLinesAndBands;
+
+        // Combine all plotBands
+        const allPlotBands = [...xAxisPlotBands, ...yAxisPlotBands];
+
+        allPlotBands.forEach((plotBand) => {
+            // We need to exit early if there is no label - the allPlotBands array
+            // will include the plotLine for the zero line as well as the plotBands
+            // for the range annotations
+            if (!plotBand.label) {
+                return;
+            }
+
+            // If the label is set to be inside the plotBand, exit early and don't draw a line
+            if (plotBand.options.label.x === undefined && plotBand.options.label.y === undefined) {
+                return;
+            }
+
+            // Get the plotBand element
+            const plotBandElement = plotBand.svgElem.element;
+            const plotBandRect = plotBandElement.getBoundingClientRect();
+
+            // Get the label element
+            const labelElement = plotBand.label.element;
+            const labelRect = labelElement.getBoundingClientRect();
+
+            // Get the label position relative to the plotBand (left, right, top or bottom)
+
```

Comment:
```suggestion
```
Just a minor coding style change to match other comments

Reply:
Updated.